base/abstractarray.jl

```diff
@@ -1916,7 +1916,6 @@ julia> foreach(x -> println(x^2), a)
 49
 ```
 """
-foreach(f) = (f(); nothing)
```

Comment:
It seems like this method is necessary to (1) resolve the ambiguity between `foreach(f, itrs...)` and `foreach(f, itrs::Tuple...)` in the empty-iterator case and (2) be non-breaking (since `for z in zip()` doesn't currently work: #43821).